content/blog/2018-02-07-update-on-async-rendering.md

```diff
@@ -0,0 +1,121 @@
+---
+title: Update on Async Rendering
+author: [bvaughn]
+---
+
+For the past few months, the React team has been experimenting with [asynchronous rendering](/blog/2017/09/26/react-v16.0.html#new-core-architecture), and we are very excited about the new features it enables.
+
+Along the way our research has shown that some of our legacy component lifecycles tend to encourage unsafe coding practices. They are:
+
+* `componentWillMount`
+* `componentWillReceiveProps`
+* `componentWillUpdate`
+
+Because of this, we have decided to rename these lifecycles—(adding an "UNSAFE_" prefix)—in a future release. The plan for this is as follows:
+
+* **16.3**: Introduce aliases for the unsafe lifecycles, `UNSAFE_componentWillMount`, `UNSAFE_componentWillReceiveProps`, and `UNSAFE_componentWillUpdate`. (Both the old lifecycle names and the new aliases will work in this release.)
+* **16.4**: Enable deprecation warning for `componentWillMount`, `componentWillReceiveProps`, and `componentWillUpdate`. (Both the old lifecycle names and the new aliases will work in this release.)
+* **17.0**: Remove `componentWillMount`, `componentWillReceiveProps`, and `componentWillUpdate` . (Only the new "UNSAFE_" lifecycle names will work in this release.)
+
+In this post, we will explore some of the potential capabilities of async rendering, and we'll outline a migration plan for components that rely on the deprecated lifecycles.
```

Comment:
should mention the codemod you added in reactjs/react-codemod#195! 😄

Reply:
I was uncertain about this, but leaning toward mentioning the codemod along with the release of 16.4. My reasoning is that 16.4 will be when the deprecation warnings are enabled, and the codemod is really only a bandaid fix for those. This release (16.3) is more focused on maintainers who want to fix their projects in the "right" way going forward (which won't be using the codemod, but rather following the recipes). So I thought mentioning it might lead people down the wrong path potentially?

Reply:
ah yeah I can totally see that. makes sense to me!